src/file/reader.rs

```diff
@@ -170,16 +171,48 @@ impl SerializedFileReader {
     for rg in t_file_metadata.row_groups {
       row_groups.push(RowGroupMetaData::from_thrift(schema_descr.clone(), rg)?);
     }
+    let column_orders =
+      Self::parse_column_orders(t_file_metadata.column_orders, &schema_descr);
 
     let file_metadata = FileMetaData::new(
       t_file_metadata.version,
       t_file_metadata.num_rows,
       t_file_metadata.created_by,
       schema,
-      schema_descr
+      schema_descr,
+      column_orders
     );
     Ok(ParquetMetaData::new(file_metadata, row_groups))
   }
+
+  /// Parses column orders from Thrift definition.
+  /// If no column orders are defined, returns `None`.
+  fn parse_column_orders(
+    t_column_orders: Option<Vec<TColumnOrder>>,
+    schema_descr: &SchemaDescriptor
+  ) -> Option<Vec<ColumnOrder>> {
+    match t_column_orders {
+      Some(orders) => {
+        // Should always be the case
+        assert_eq!(orders.len(), schema_descr.num_columns(),
+          "Column order length mismatch");
+        let mut res = Vec::new();
+        for (i, column) in schema_descr.columns().iter().enumerate() {
+          match orders[i] {
+            TColumnOrder::TYPEORDER(_) => {
+              let sort_order = ColumnOrder::get_sort_order(
```

Comment:
The reason I added method to column order was emphasis that this is a column order feature. When adding to column descriptor, we should consider adding actual `ColumnOrder` to it, because having method `get_sort_order` might be a bit confusing, people would want to get sort order, but maybe it is a format that does not support column order, so the actual sort order might be different from the one returned by the method.

I was thinking about adding `ColumnOrder` field to `ColumnDescriptor`. But this will require some additional changes, that I can make in a separate PR. Let me know if you think it is a good idea.

Reply:
I see. The current approach is fine too. Let's keep it this way then.